sympy/core/evalf.py

```diff
@@ -871,7 +871,31 @@ def evalf_atan(v, prec, options):
 
 
 def evalf_subs(prec, subs):
-    """ Change all Float entries in `subs` to have precision prec. """
+    """
+    Change all Float entries in `subs` to have precision prec.
+    Note:
+    It is always suggested to use expr.evalf(subs=...) over,
+    expr.subs(...).evalf() or expr.evalf().subs(...) because of the
+    following reasons:-
+    i.  The subs dictionary tells the evalf algorithms which symbols should be
+        replaced with numbers when they are encountered.
+    ii. expr.evalf(subs=...) the expression is run through the evalf algorithm,
+        which takes into account various issues that can lead to loss of
+        significance;
+    iii.expr.evalf(subs=...) avoids the loss of significance caused due to
+        naive substitution.
+    Examples
+    ========
+        from sympy.core.evalf import evalf
+        from sympy.abc import x,y,z
+        (x+y-z).subs({x:1e100,y:1,z:1e100})
+        0
+        (x+y-z).evalf(subs={x: 1e100, y: 1, z: 1e100})
```

Comment:
The code examples in the documentation should be written in Python interactive prompt. For example, all these lines should be written as: 
```
>>> from sympy.core.evalf import evalf
>>> from sympy.abc import x,y,z
>>> (x+y-z).subs({x:1e100,y:1,z:1e100})
0
>>> (x+y-z).evalf(subs={x: 1e100, y: 1, z: 1e100})
1.0000000000000 
```
You can look into the other files for examples.